packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -484,7 +501,9 @@ export const useField = <
     onChange: handleInputChange,
     onKeyDown: handleInputKeyDown,
     onMouseUp: handleInputMouseUp,
+    onClear: handleClearValue,
     error: inputError,
     ref: handleRef,
+    clearable: Boolean(clearable && inputHasFocus && !areAllSectionsEmpty),
```

Comment:
Based on the previous discussions, I believe that we were closing in on having an implementation very similar to the one, that Autocomplete has, where `hovering` would also show the clearable button